icefall/decode.py

```diff
@@ -740,16 +748,15 @@ def rescore_with_whole_lattice(
                 "is too large, or the input sound file is difficult to "
                 "decode, you will meet this exception."
             )
-
-            # NOTE(fangjun): The choice of the threshold 1e-9 is arbitrary here
-            # to avoid OOM. You may need to fine tune it.
-            inv_lattice = k2.prune_on_arc_post(inv_lattice, 1e-9, True)
+            inv_lattice = k2.prune_on_arc_post(
+                inv_lattice,
+                prune_th_list[loop_count],
```

Comment:
When `loop_count` is 0, it will cause access-out-of boundary error as `prune_th_list` has only 10 elements.

Reply:
I have moved `loop_count += 1` to the end of the while loop. https://github.com/wgb14/icefall/blob/ff88356ee41f900e470dafa7fe1cc9b5bbaf95a3/icefall/decode.py#L759
And when `loop_count` gets to 10, `return None` would be called before this line.
https://github.com/wgb14/icefall/blob/ff88356ee41f900e470dafa7fe1cc9b5bbaf95a3/icefall/decode.py#L737-L741
So only `prune_th_list[0:9]` are accessed.

Reply:
Ah, I see.